lib/ssh/test/ssh_connection_SUITE.erl

```diff
@@ -1175,6 +1179,59 @@ do_start_shell_exec_fun(Fun, Command, Expect, ExpectType, Config) ->
     ssh:close(ConnectionRef),
     ssh:stop_daemon(Pid).
 
+%%--------------------------------------------------------------------
+%% Issue GH-8223
+trap_exit_connect(Config) when is_list(Config) ->
+    PrivDir = proplists:get_value(priv_dir, Config),
+    UserDir = filename:join(PrivDir, nopubkey),
+    file:make_dir(UserDir),
+    SysDir = proplists:get_value(data_dir, Config),
+    {Pid, Host, Port} = ssh_test_lib:daemon([{system_dir, SysDir},
+                                             {user_dir, UserDir},
+                                             {password, "morot"}]),
+    %% Fake an EXIT message
+    ExitMsg = {'EXIT', self(), make_ref()},
+    self() ! ExitMsg,
+
+    {ok, ConnectionRef} = ssh:connect(Host, Port, [{silently_accept_hosts, true},
+                                                   {save_accepted_host, false},
+                                                   {user, "foo"},
+                                                   {password, "morot"},
+                                                   {user_interaction, true},
+                                                   {user_dir, UserDir}]),
+    ssh:close(ConnectionRef),
+    ssh:stop_daemon(Pid),
+
+    %% Ensure the EXIT message is still there
+    receive
+        ExitMsg -> ok
+    after 0 ->
+        ct:fail("No EXIT message")
+    end.
+
+%%--------------------------------------------------------------------
+%% Issue GH-8223
+trap_exit_daemon(Config) when is_list(Config) ->
```

Comment:
Have you seen `trap_exit_daemon` testcase failing on clean master branch? In my execution it always succeeds.
By reading execution traces I don't thin 'EXIT' issue occurs for `ssh:daemon` calls

Reply:
I don't really remember, but I don't think it did, and we probably added this test just to be sure 🤷🏼‍♀️ I'm on vacation ATM, reading code on my tiny mobile is difficult, the codebase is complex... tl;dr, reasoning about whether or not this _could_ happen is pretty hard for me right now 😅